### PR TITLE
rpc: populate allowed/not allowed node ids when listing orders

### LINKED ISSUE #428: rpc: allowed/not Allowed node ids not populated when listing orders.
The fields are stored properly in the db and properly taken into account during the matches but the `order list` rpc method does not populate the values in the response.

```
pool o s bid  --allowed_node_id=${ALLOWED_NODE_ID} ...
// submission succeeds

pool o l | jq .bids[].allowed_node_ids
// returns null
```

### PR DESCRIPTION
**Pull request by positiveblue**
Fixes #428 
Depends on #427

### CODE REVIEW

**Review by guggero — CHANGES_REQUESTED** [review]

File: rpcserver.go [hunk 1]
```diff
@@ -1698,6 +1698,20 @@ func (s *rpcServer) ListOrders(ctx context.Context,
 			}
 		}
 
+		allowedNodeIDs := make(
+			[][]byte, 0, len(dbOrder.Details().AllowedNodeIDs),
+		)
+		for _, nodeID := range dbOrder.Details().AllowedNodeIDs {
+			allowedNodeIDs = append(allowedNodeIDs, nodeID[:])
```
[the remark this answers is not in the document]
Comment (positiveblue) [reply rpcserver.go]: Right, tests added for it :+1:

To have this done everywhere with the same structure we could have a `serialize.go` in each package that has the logic to serialize/deserialize obj from/to rpc and `rpcserver.go` calls those. In that way we can keep the testing (with invalid inputs) in the package itself. Similar to the current `rpc_parse.go` in orders but without splitting the logic between that file and the `rpcserver.go`.

File: order/rpc_parse.go [hunk 2]
```diff
@@ -524,6 +520,43 @@ func ParseRPCSign(signMsg *auctioneerrpc.OrderMatchSignBegin) (AccountNonces,
...
+		copy(nodeID, nodeIDs[i][:])
+
+		res = append(res, nodeID)
+	}
+
+	return res
+}
+
+// UnmarshalNodeIDSlice returns a slice of node ids from a flatten version.
+func UnmarshalNodeIDSlice(slice [][]byte) ([][33]byte, error) {
```
Comment (guggero) [inline order/rpc_parse.go:539]: Nice! We can re-use this on the server side as well.

File: order/rpc_parse.go [hunk 3]
```diff
@@ -524,6 +520,43 @@ func ParseRPCSign(signMsg *auctioneerrpc.OrderMatchSignBegin) (AccountNonces,
 	return nonces, prevOutputs, nil
 }
 
+// MarshalNodeIDSlice returns a flatten version of an slice of node ids to be
```
Comment (guggero) [inline order/rpc_parse.go:523]: nit: s/flatten/flattened/

**Review by guggero — APPROVED** [review]

File: tools/Dockerfile [hunk 4]
```diff
@@ -11,6 +11,9 @@ RUN cd /tmp \
   && mkdir -p /tmp/build/.modcache \
   && cd /tmp/tools \
   && go install -trimpath -tags=tools github.com/golangci/golangci-lint/cmd/golangci-lint \
-  && chmod -R 777 /tmp/build/
+  && chmod -R 777 /tmp/build/ \
+  && git config --global --add safe.directory /build 
+  # The last line is needed to ensure that go build is able to gather
```
Comment (guggero) [inline tools/Dockerfile:16]: Can be removed.

File: tools/Dockerfile [hunk 5]
```diff
@@ -11,9 +11,6 @@ RUN cd /tmp \
   && mkdir -p /tmp/build/.modcache \
   && cd /tmp/tools \
   && go install -trimpath -tags=tools github.com/golangci/golangci-lint/cmd/golangci-lint \
-  && chmod -R 777 /tmp/build/ \
-  && git config --global --add safe.directory /build 
-  # The last line is needed to ensure that go build is able to gather 
-  # information from the vsc used in the builds to get the commit hash. 
+  && chmod -R 777 /tmp/build/
```
Comment (positiveblue) [inline tools/Dockerfile:14]: reverts a commit that I pushed by error in #426

**Review by jbrill — APPROVED** [review]
LGTM 👍 very comprehensive tests. I was unable to get the fields showing in an end to end test as it looks like there's some additional field marshaling we need to do subasta-side.

**Pull request merged**